test/pools/integration/BatchingAndPayment.t.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.28;
+
+import "test/pools/integration/BaseTest.t.sol";
+
+contract TestBatchingAndPayment is BaseTest {
+    /// forge-config: default.isolate = true
+    function testExecuteNoSendNoPay() public {
+        vm.prank(ADMIN);
+        PoolId poolId = guardian.createPool(FM, USD, multiShareClass);
+
+        vm.startPrank(FM);
+
+        bytes[] memory cs = new bytes[](1);
+        cs[0] = abi.encodeWithSelector(poolRouter.setPoolMetadata.selector, "");
+
+        poolRouter.execute(poolId, cs);
+
+        // Check no messages were sent as intended
+        assertEq(cv.messageCount(), 0);
+    }
+
+    /// forge-config: default.isolate = true
+    function testExecuteSendNoPay() public {
+        vm.prank(ADMIN);
+        PoolId poolId = guardian.createPool(FM, USD, multiShareClass);
+
+        vm.startPrank(FM);
+
+        bytes[] memory cs = new bytes[](1);
+        cs[0] = abi.encodeWithSelector(poolRouter.notifyPool.selector, CHAIN_CV);
+
+        vm.expectRevert(bytes("Gateway/not-enough-gas-funds"));
+        poolRouter.execute(poolId, cs);
+    }
+
+    /// Test the following:
+    /// - multicall()
+    ///   - execute(poolA)
+    ///      - notifyPool()
+    ///   - execute(poolA)
+    ///      - notifyPool()
+    ///
+    /// will pay only for one message. The batch sent is [NotifyPool, NotifyPool].
```

Comment:
Doesn't this comment contradict paying `GAS * 2` below?

Reply:
I forgot to fix it! But I would do in a follow up PR